base_geoengine/static/src/js/views/geoengine/records_panel/records_panel.esm.js

```diff
@@ -0,0 +1,77 @@
+/** @odoo-module */
+
+/**
+ * Copyright 2023 ACSONE SA/NV
+ */
+import {useService} from "@web/core/utils/hooks";
+import {SearchBarRecords} from "./search_bar_records/search_bar_records.esm";
+
+const {Component, useState, onWillRender, onWillStart, onWillUpdateProps} = owl;
+
+export class RecordsPanel extends Component {
+    setup() {
+        super.setup();
+
+        this.state = useState({
+            isFolded: false,
+            isClicked: 0,
+            modelDescription: "",
+            records: [],
+        });
+        this.orm = useService("orm");
+        onWillStart(() => (this.state.records = this.props.list.records));
+        onWillUpdateProps((nextProps) => (this.state.records = nextProps.list.records));
+        onWillRender(async () => {
+            // Retrieves the name of the current model
+            const result = await this.orm.call("ir.model", "display_name_for", [
```

Comment:
be aware that the `onWillRender` hook is not asynchronous, so you will have a small flickering issue: owl will render immediately, then when the rpc comes back, it will render again